src/GafferUI/StandardStyle.cpp

```diff
@@ -1283,18 +1283,14 @@ Imath::Color3f StandardStyle::colorForState( Color c, State s, const Imath::Colo
 	}
 	else if( s == Style::DisabledState )
 	{
-		result = lerp(
-			result, Color3f( 0.28, 0.28, 0.28 ),
-			c == ConnectionColor ? 0.8 : 0.75
-		);
-		// if( c == ConnectionColor )
-		// {
-		// 	result = Color3f( 0.2f );
-		// }
-		// else
-		// {
-		// 	result = result * 0.25f + 0.75f * Color3f( 0.28, 0.28, 0.28f );
-		// }
+		if( c == ConnectionColor )
+		{
+			result = lerp( 1.0f, 0.26f / luminance( result ), 0.85f ) * result;
```

Comment:
Do we need to worry about input where `luminance( result ) < 0.26`, or worse `luminance( result ) == 0`?

Reply:
`luminance( result ) < 0.26` seems fine - it would compute a multiplier > 1 which would increase the dark color to be closer than grey when disabled, which seems reasonable.

`== 0` is a problem though, since you can't multiply it up.  I should probably just toss a clamp in.

Reply:
In an attempt to get folks at IE happy, I've now come up with yet another tweak on this.

I now desaturate partway, then fix the luminance to 0.31 ( slightly brighter than background ).  This seems to be a bit more consistent in making sure that different colors of noodles are visible when disabled, rather than relying on the original noodle color being different enough in luminance that it's still brighter after being blended heavily towards the background.

It works on black noodles, but is ugly looking ( it's weird that they switch from darker than background to brighter ), but that's probably OK - we're not really expecting dark noodles, as long as it's not a divide-by-zero like before?

Reply:
It feels a bit odd to me that noodle colours would be sufficiently more important than node colours that they warrant different handling. And the black noodles thing seems less than ideal too. But it appears to work just fine for vanilla Gaffer, so if it's what we need to get this out of the door, so be it.

Reply:
Current version is 9e1d125a1